nifpga/bitfile.py

```diff
@@ -32,12 +32,15 @@ def __init__(self, filepath, parse_contents=False):
             if reg.datatype is not None:
                 assert reg.name not in self._registers, \
                     "One or more registers have the same name '%s', this is not supported" % reg.name
-                self._registers[reg.name] = reg
+                if not reg.is_array() and reg.datatype is not DataType.Fxp:
+                    self._registers[reg.name] = reg
```

Comment:
I think this still breaks.  If the register is an array, the conditional will always be "if false and <doesn't matter>:  Maybe instead something like:

    # we don't yet support FXP Arrays
    if not (reg.is_array() and reg.datatype is DataType.Fxp):

Reply:
Gah, should be fixed now.

Reply:
Did none of these erroneous code snippets cause a test to fail? Maybe you should add some tests that would have failed.